comp/core/workloadmeta/collectors/catalog-less/catalog.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package catalog is a wrapper that loads workloadmeta collectors, while having less
+// than the full set. Currently only used by the dogstatsd binary, this catalog does
+// not include the process-collector due to its increased dependency set.
+package catalog
+
+import (
+	"go.uber.org/fx"
+
+	remoteworkloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/collectors/internal/remote/workloadmeta"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+)
+
+// GetCatalog returns the set of FX options to populate the catalog
+func GetCatalog() fx.Option {
```

Comment:
can we do something like

```golang
func GetCatalog(collectors ...CollectorType) fx.Option {
  // if collectors empty add all collectors. else only appends to the list the one that are return by `getCollectorOptions()`
}
```

The idea is to have to create a many `GetCatalog()` function has we have agent processes.

Reply:
I had thought your stance was that collectors should remain an implementation detail that is not visible to users. Do we need to allow that level of configuration, that users can pass in the list of collectors they want?

The advantage of doing it this way, where the `GetCatalog` call is in its own package (and soon, in its own component implementation) is that the build of the agent doesn't pull in unnecessary dependencies. This PR began because the dogstatsd binary fails in CI if it adds unwanted deps. This solution leverages a common pattern that components has used recently: make the implementation choice by importing what is needed, which lets us explicitly control the dependency graph.